intelmq/bots/outputs/rpz_file/output.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+import datetime
+from datetime import datetime
+import os
+from collections import defaultdict
+from pathlib import Path
+
+from intelmq.lib.bot import OutputBot
+
+RPZ_INDICATOR_MAP = {
+    "source.fqdn": "Intel::DOMAIN"
+}
+
+now = datetime.now()
+
+
+class RpzFileOutputBot(OutputBot):
+    _file = None
+    format_filename: bool = False
+    hierarchical_output: bool = False
+    keep_raw_field: bool = False
+    message_jsondict_as_string: bool = False
+    message_with_type: bool = False
+    single_key: bool = False
+    is_multithreadable = False
+
+    encoding_errors_mode = 'strict'
+    file: str = "/opt/intelmq/var/lib/bots/file-output/rpz"
+
+    cname: str = ""
+    organization_name: str = ''
+    rpz_domain: str = ''
+    hostmaster_rpz_domain: str = ''
+    rpz_email: str = ''
+    ttl: str = '3600'
+    generate_time: str = now.strftime("%Y-%m-%d %H:%M:%S")
+    ncachttl: str = '60'
+    serial: str = now.strftime("%y%m%d%H%M")
+    refresh: str = '60'
+    retry: str = '60'
+    expire: str = '432000'
+    test_domain: str = ''
+
+    rpz_header = ""
+
+    def init(self):
+        self._file = None
+
+        self.set_rpz_header()
+
+        self.logger.debug("Opening %r file.", self.file)
+        self.format_filename = getattr(self, 'format_filename', False)
+        self.errors = getattr(self, 'encoding_errors_mode', 'strict')
+        if not self.format_filename:
+            self.open_file(self.file)
+        self.logger.info("File %r is open.", self.file)
+
+    def set_rpz_header(self):
+        self.rpz_header = "$TTL " + str(self.ttl) + "\n@ SOA " + self.rpz_domain + ". " + self.hostmaster_rpz_domain + ". " + str(self.serial) + " " + str(self.refresh) + " " + str(self.retry) + " " + str(self.expire) + " " + str(self.ncachttl) + "\n NS localhost.\n;\n; " + self.organization_name + " Response Policy Zones (RPZ)\n; Last updated: " + str(self.generate_time) + " (UTC)\n;\n; Terms Of Use: https://" + self.rpz_domain + "\n; For questions please contact " + self.rpz_email + "\n;\n"
```

Comment:
Please format this string by other means to make it more readable, e.g. using format-strings, like
```python
self.rpz_header = f"""$TTL {self.ttl}
@ SOA {self.rpz_domain}. {self.hostmaster_rpz_domain}. {self.serial}"""
```
etc.